libraries/legacy/form/field/componentlayout.php

```diff
@@ -65,7 +65,7 @@ protected function getInput()
 		if ($this->form instanceof JForm)
 		{
 			$template_style_id = $this->form->getValue('template_style_id');
-			$template_style_id = preg_replace('#\W#', '', $template_style_id);
+			$template_style_id = preg_replace('#\W#', '', (string) $template_style_id);
```

Comment:
```suggestion
			$template_style_id = preg_replace('#\W#', '', $template_style_id);
```

That should not be a string but an ID in the first place right?